### DISCUSSION TITLE
Custom Overlay not positioned correctly

### DISCUSSION THREAD

**Discussion by emaborsa** [visgl/react-google-maps#626] [Q&A]
I am working with React, Typescript and Google Maps CustomOverlay. The idea is to allow the user to draw lines on the map, [here a Sandbox](https://codesandbox.io/p/sandbox/y7f7wy).

The problem is that the canvas is rendered with the top-left corner in the middle of the map:
![gm_canvas](https://github.com/user-attachments/assets/04afd3d4-2a2a-443a-806d-b4cf0b4c55a3)

And so the user cant draw line in the left or upper side of the map:
![03 12 2024_13 08 34_REC](https://github.com/user-attachments/assets/228cef0c-b4ee-4723-8029-499e150dd729)

I searched on the web but did not find a solution. Any hint to stretch the overlay over the whole (displayed) map?

**Comment by usefulthink** [comment]
First off, it looks like you're in the wrong project here :)
You should probably post this question here instead: https://github.com/JustFly1984/react-google-maps-api

The origin point of the `overlayLayer` pane is always in the center of the map when resting. If you want to change that, you can translate your canvas-element using `canvas.style.transform = 'translate(-50%, -50%)'` for example. See here: https://codesandbox.io/p/sandbox/crazy-sinoussi-2zgp53

As a hint, doing something like this with the `OverlayView` will get really complicated really fast. Since the `OverlayView` itself isn't anchored geographically (it doesn't have latitude/longitude), so it jumps around when you move the map. When the map is moved, you can either move the canvas so it stays where it belongs or you have to update everything within the canvas to cancel out map-movement.

If all you need is simple lines or polygons, have a look at the google.maps.Polyline and google.maps.Polygon classes (those can be set to "editable", see here: https://developers.google.com/maps/documentation/javascript/examples/user-editable-shapes) or the `google.maps.drawing` library (https://developers.google.com/maps/documentation/javascript/examples/drawing-tools).

**Comment by emaborsa** [comment]
Oh, I apologize. I will move to that repo.
Thanks.

**Comment by naqshbandhamza** [comment]
> I am working with React, Typescript and Google Maps CustomOverlay. The idea is to allow the user to draw lines on the map, [here a Sandbox](https://codesandbox.io/p/sandbox/y7f7wy).
> 
> The problem is that the canvas is rendered with the top-left corner in the middle of the map: ![gm_canvas](https://private-user-images.githubusercontent.com/9129912/392019568-04afd3d4-2a2a-443a-806d-b4cf0b4c55a3.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTcxMDMzNDAsIm5iZiI6MTc1NzEwMzA0MCwicGF0aCI6Ii85MTI5OTEyLzM5MjAxOTU2OC0wNGFmZDNkNC0yYTJhLTQ0M2EtODA2ZC1iNGNmMGI0YzU1YTMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDkwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA5MDVUMjAxMDQwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWI0ZDQ4MjQ1N2FkZThlMGU4Y2UxNzNiZDU5OWRlYTIzZTg5MDQ5YjQ0MGQ1MDM4MDdmZWVlMzZjYmFjOGU0NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.brvxhitPzk2o4vfbH7EQy20rZDd7ommINwlFcHtJ_A0)
> 
> And so the user cant draw line in the left or upper side of the map: ![03 12 2024_13 08 34_REC](https://private-user-images.githubusercontent.com/9129912/392017349-228cef0c-b4ee-4723-8029-499e150dd729.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTcxMDMzNDAsIm5iZiI6MTc1NzEwMzA0MCwicGF0aCI6Ii85MTI5OTEyLzM5MjAxNzM0OS0yMjhjZWYwYy1iNGVlLTQ3MjMtODAyOS00OTllMTUwZGQ3MjkuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDkwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA5MDVUMjAxMDQwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2ZkMjE4YTk2NzlhNDZkNzEyYTVhMTk2OTFlMmE2OWRkM2M4MjcyY2Y2M2U4MDkzYmJlMzU5NjMxY2RjNGEwMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.7BXxcpVXRTuAa1tHWpMAr90LA-sne_xBVjCpALhE5pk) [ ![03 12 2024_13 08 34_REC](https://private-user-images.githubusercontent.com/9129912/392017349-228cef0c-b4ee-4723-8029-499e150dd729.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTcxMDMzNDAsIm5iZiI6MTc1NzEwMzA0MCwicGF0aCI6Ii85MTI5OTEyLzM5MjAxNzM0OS0yMjhjZWYwYy1iNGVlLTQ3MjMtODAyOS00OTllMTUwZGQ3MjkuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDkwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA5MDVUMjAxMDQwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2ZkMjE4YTk2NzlhNDZkNzEyYTVhMTk2OTFlMmE2OWRkM2M4MjcyY2Y2M2U4MDkzYmJlMzU5NjMxY2RjNGEwMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.7BXxcpVXRTuAa1tHWpMAr90LA-sne_xBVjCpALhE5pk) ](https://private-user-images.githubusercontent.com/9129912/392017349-228cef0c-b4ee-4723-8029-499e150dd729.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTcxMDMzNDAsIm5iZiI6MTc1NzEwMzA0MCwicGF0aCI6Ii85MTI5OTEyLzM5MjAxNzM0OS0yMjhjZWYwYy1iNGVlLTQ3MjMtODAyOS00OTllMTUwZGQ3MjkuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDkwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA5MDVUMjAxMDQwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2ZkMjE4YTk2NzlhNDZkNzEyYTVhMTk2OTFlMmE2OWRkM2M4MjcyY2Y2M2U4MDkzYmJlMzU5NjMxY2RjNGEwMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.7BXxcpVXRTuAa1tHWpMAr90LA-sne_xBVjCpALhE5pk) [ ](https://private-user-images.githubusercontent.com/9129912/392017349-228cef0c-b4ee-4723-8029-499e150dd729.gif?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTcxMDMzNDAsIm5iZiI6MTc1NzEwMzA0MCwicGF0aCI6Ii85MTI5OTEyLzM5MjAxNzM0OS0yMjhjZWYwYy1iNGVlLTQ3MjMtODAyOS00OTllMTUwZGQ3MjkuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDkwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA5MDVUMjAxMDQwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2ZkMjE4YTk2NzlhNDZkNzEyYTVhMTk2OTFlMmE2OWRkM2M4MjcyY2Y2M2U4MDkzYmJlMzU5NjMxY2RjNGEwMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.7BXxcpVXRTuAa1tHWpMAr90LA-sne_xBVjCpALhE5pk)
> 
> I searched on the web but did not find a solution. Any hint to stretch the overlay over the whole (displayed) map?

Hello!!! I know its quite late but i am facing the very same issue were u able to find a way? the thing is I tried to translate my canvas to (-50%,-50%) which did make it cover the whole map but when you do that; then you cannot maintain the position of your drawings. they drift away on every zoom (not persisting on the very location /landmark you first placed them)
